core/keyboard_nav/ast_node.js

```diff
@@ -293,7 +293,7 @@ Blockly.ASTNode.prototype.findNextForField_ = function() {
   var location = /** @type {!Blockly.Field} */ (this.location_);
   var input = location.getParentInput();
   var block = location.getSourceBlock();
-  var curIdx = block.inputList.indexOf(input);
+  var curIdx = block.inputList.indexOf(/** @type {!Blockly.Input} */ (input));
```

Comment:
Why isn't the result of getParentInput already of type Blockly.Input?

Reply:
It is of type Blockly.Input but not !Blockly.Input. We might be able to change that though? I can't think of a situation where the field is attached to a block, but doesn't have a parent input?

Reply:
Yeah that's imposible.

Reply:
ok! well looking at this again it looks like getParentInput should still return null in the case that the sourceBlock_ is null so I don't think we can return !Blockly.Input from there.